lib/main.dart

```diff
@@ -1,12 +1,112 @@
 import 'package:flutter/material.dart';
+import 'data.dart';
 
 void main() => runApp(MyApp());
 
 class MyApp extends StatelessWidget {
-  @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      home: Center(child: Text('Empty Shell for Art Museum')),
+      theme: ThemeData(primaryColor: Colors.teal[50]),
+      home: HomePage(),
+    );
+  }
+}
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  List<ObjectInfo> picData;
+
+  void initState() {
+    super.initState();
+    requestImages().then((images) {
+      setState(() => picData = images);
+    });
+  }
+
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.teal[100],
+      appBar: AppBar(
+        title: Text('The Art Museum'),
+        backgroundColor: Colors.teal[50],
+      ),
+      body: Column(
+        children: [
+          Expanded(child: worksOfArt()),
+          RaisedButton(
+            child: Text('Load more Artwork'),
+            onPressed: () async {
+              final images = await requestImages();
+              setState(() => picData.addAll(images));
+            },
+            color: Colors.white,
+            splashColor: Colors.brown,
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget worksOfArt() {
+    final deviceSize = MediaQuery.of(context).size;
+    return ListView.builder(
+      scrollDirection: Axis.horizontal,
+      itemCount: (picData == null) ? 0 : picData.length,
+      itemBuilder: (context, index) {
+        var data = picData[index];
+        var imageData = picData[index].images[0];
+        var height, width;
+        if (imageData.height / 3 > deviceSize.height - 245) {
```

Comment:
Is there a better way we can determine how to scale the image? It looks like some images exceed the size of the viewport, while others are scaled not to. What is the expected behavior/bounds for the image size?

Within viewport:
![Screen Shot 2019-08-01 at 10 42 52 AM](https://user-images.githubusercontent.com/16964204/62315004-4123fa80-b449-11e9-923c-0c2f087adcc3.png)

Outside of viewport:
![Screen Shot 2019-08-01 at 10 43 21 AM](https://user-images.githubusercontent.com/16964204/62314998-3f5a3700-b449-11e9-930c-580356481a08.png)